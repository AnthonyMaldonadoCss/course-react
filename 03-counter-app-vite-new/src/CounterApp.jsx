import { useState } from 'react';
import PropTypes from 'prop-types';



export const CounterApp = ({ value }) => {
  const [ counter, setCounter ] = useState( value );

  const handleClick = (event) => {
    console.log(value);
    const { action } = event.target.dataset;
    console.log( 'Hola mundo' )
    console.log( event );
    console.log( event.target.dataset.action );

    switch ( action ) {
      case 'add':
        setCounter( counter + 1 );
        break;
      case 'substract':
        setCounter( counter - 1 );
        break;
      case 'reset':
        setCounter( value );
        break;
      default:
        setCounter( value );
        break;
    }
  
    // setCounter( counter + 1 );
    // setCounter( (c) => c + 1 );
  
  }

  return (
    <>
      <h1> CounterApp </h1>
      <h2> { counter } </h2>
      <button onClick={ handleClick } data-action='add'> +1 </button> &nbsp;
      <button onClick={ handleClick } data-action='substract'> -1 </button> &nbsp;
      <button onClick={ handleClick } data-action='reset'> Reset </button> 
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
  value: 0
}