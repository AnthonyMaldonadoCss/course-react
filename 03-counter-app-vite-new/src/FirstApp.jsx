import PropTypes from 'prop-types';


const nombre = 'Anthony Maldonado';
const date = new Date();

const numeros = [1,2,3,4,5,6,7,8,9,10];

const loop = () => {
  const li = [];
  for (let [i,numero] of numeros.entries()) {
    li.push( <li key={i} > {numero} </li> )
  }
  return li
}
/**ES MEJOR DECLARAR LAS FUNCIONES FUERA PORQUE AL MOMENTO DE RENDERIZAR 
 * SE RE EVALURAN LAS FUNCIONES, CONSTANTES, ETC...
*/

export const FirstApp = ({title, subTitle}) => {
  return (
    <> 
      <h1 style={ { color: "red" } } > 
        { nombre }: 
        <span style={ { color: "blue", fontSize:"xx-large" } }> 
          &nbsp; Software Developer 
        </span>
      </h1> 

      <h2>
        { title }
      </h2>

      <h4 id='subtitle'>
        { subTitle } 
      </h4>
      
      <p> { date.getUTCDate() } - { date.getUTCMonth() } - { date.getFullYear() } </p>
      
      <p> El fragment es para devolver lo que se quiere renderizar en un elemento padre </p>
      
      <ul style={ { display: "flex", flexDirection: "row", justifyContent:"space-evenly", alignItems: "flex-start"} }>
        <> { loop() } </>
      </ul>
    
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  title: 'Hola, soy Anthony',
  subTitle: '123'
}