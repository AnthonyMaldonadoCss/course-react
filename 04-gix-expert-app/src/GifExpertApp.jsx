import { useState } from 'react'
import GifGrid  from './components/GifGrid';
import  AddCategory  from './components/AddCategory';

const GifExpertApp = () => {
  
  const [categories, setCategories] = useState(['Dragon Ball']);

  const validateCategory = (newCategory) => {
    
    const validate = (category) => {
      const bodyCategory = category.split(' ').map(word => word.toLowerCase()).join(' ');
      newCategory = newCategory.split(' ').map(word => word.toLowerCase()).join(' ');
      return bodyCategory === newCategory
    }

    return categories.some( validate );
  }

  const onAddCategory = (NewCategory) => {
    if ( validateCategory(NewCategory) ) return;
    
    setCategories([ ...categories, NewCategory ]);
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
        {
          categories.map((category) => (
            <GifGrid 
              key={ category } 
              category={ category } 
            />
          ))
        }
    </>
  )
}

export default GifExpertApp
