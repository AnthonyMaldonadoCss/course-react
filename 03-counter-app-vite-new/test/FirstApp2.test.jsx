import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => {

  const title = 'Hola, soy Anthony'
  const subtile = 123;
  
  test('Debe de hacer match con el snapShot', () => {
    const { container } = render(<FirstApp title={ title } subTitle={ subtile } />)
    expect( container ).toMatchSnapshot(); 
  })

  test('Debe de mostrar el mensaje, Hola, soy Anthony', () => {
    render( <FirstApp title={ title } subTitle={ subtile } /> )
    expect( screen.getAllByText(title) ).toBeTruthy();
  })

  test('Debe mostrar el titulo en un H1', () => {
    render( <FirstApp title={ title } subTitle={ subtile } /> );
    
    expect( 
      screen.getByRole('heading', { level: 2 }).innerHTML 
    ).toContain( title );
  })

  test('Debe mostrar el subtitulo en un H4', () => {
    render( <FirstApp title={ title } subTitle={ subtile } /> );

    expect( 
      screen.getByRole('heading', { level: 4 }).innerHTML 
    ).toContain( String(subtile) );
  })
  
})
