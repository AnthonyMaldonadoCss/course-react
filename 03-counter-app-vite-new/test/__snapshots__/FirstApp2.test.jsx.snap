// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapShot 1`] = `
<div>
  <h1
    style="color: red;"
  >
    Anthony Maldonado
    :
    <span
      style="color: blue; font-size: xx-large;"
    >
      Â  Software Developer
    </span>
  </h1>
  <h2
    data-testid="test-title"
  >
    Hola, soy Anthony
  </h2>
  <h4
    id="subtitle"
  >
    123
  </h4>
  <p>
     
    28
     - 
    3
     - 
    2024
     
  </p>
  <p>
     El fragment es para devolver lo que se quiere renderizar en un elemento padre 
  </p>
  <ul
    style="display: flex; flex-direction: row; justify-content: space-evenly; align-items: flex-start;"
  >
     
    <li>
       
      1
       
    </li>
    <li>
       
      2
       
    </li>
    <li>
       
      3
       
    </li>
    <li>
       
      4
       
    </li>
    <li>
       
      5
       
    </li>
    <li>
       
      6
       
    </li>
    <li>
       
      7
       
    </li>
    <li>
       
      8
       
    </li>
    <li>
       
      9
       
    </li>
    <li>
       
      10
       
    </li>
     
  </ul>
</div>
`;
